Add transformations from kafka topic names to hive tables names

serssp opened the pull request:
confluentinc/kafka-connect-hdfs#155

serssp commented:
[clabot:check]

cricket007 commented:
Is it not possible to do this with `RegexRouter` already?

serssp commented:
> Is it not possible to do this with `RegexRouter` already?

Unfortunately not, I tried to do it, but I received the NPE just as in this task https://github.com/confluentinc/kafka-connect-hdfs/issues/155#issuecomment-386227778

cricket007 commented:
I see. One comment I had was the default property - should that be `[-.]`? And are there unit tests that could be added for this feature?

serssp commented:
Hello, @cricket007 !

Thank you for review

I have added test.
Also I have added HiveMetaStoreException when topic name not matched. This behavior more safe and predictive.

I didn't make changes in replacement from "[.-]" to "_".  This replacement is performed after regex matching.

cricket007 commented on hive/src/test/java/io/confluent/connect/storage/hive/HiveMetaStoreTableNameConverterFactoryTest.java:
```diff
@@ -0,0 +1,62 @@
+    return Arrays.asList(new Object[][]{
+            {HIVE_TABLE_PATTERN_DEFAULT, "topic_name", "topic_name", null},
+            {HIVE_TABLE_PATTERN_DEFAULT, "topic-name", "topic_name", null},
+            {HIVE_TABLE_PATTERN_DEFAULT, "topic-name", "topic_name", null},
```
I assume one of these should be `topic.name`?

serssp replied:
Yes, thx!

cricket007 commented on hive/src/test/java/io/confluent/connect/storage/hive/HiveMetaStoreTableNameConverterFactoryTest.java:
```diff
@@ -0,0 +1,62 @@
+  public void test() {
+    HiveConfig hiveConfig = new HiveConfig(new HashMap<String, String>() {{
+      put(HIVE_TABLE_PATTERN_CONFIG, hivePatternConfig);
+    }});
```
Double bracket initialization is considered an anti-pattern in some contexts. I would say just make the HashMap and call `.put`
